rust/datafusion/src/physical_plan/hash_aggregate.rs

```diff
@@ -776,6 +785,14 @@ pub(crate) fn create_group_by_values(
     for i in 0..group_by_keys.len() {
         let col = &group_by_keys[i];
         match col.data_type() {
+            DataType::Float32 => {
```

Comment:
I've updated PR, forget about this place. Testing it with real DB example.

Reply:
I suggest adding an end-to-end sql test in https://github.com/apache/arrow/blob/master/rust/datafusion/tests/sql.rs to make sure the plumbing is all hooked up correctly

Reply:
@alamb I've added tests, but I found a strange bug probably with count.

https://github.com/apache/arrow/pull/9175/commits/012c1ac0151878f43d60e3a7090f5b99947e3294#diff-4b06103cf2132b1ab297fbe8cd42622ecbe1109ea26df7d8b358fa36d739549cR357

Thanks